include/rocksdb/advanced_options.h

```diff
@@ -132,6 +133,9 @@ struct CompressionOptions {
   // Default: 1.
   uint32_t parallel_threads;
 
+  // Options for custom compressors
+  std::unordered_map<std::string, std::string> custom_options;
+
```

Comment:
If possible, it would be better to use the Options/OptionTypeInfo system to define specific options for the extended types.  I can work with you to get that set up.